File: my-folder/0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int lt = 0;
        int rt = height.size()-1;
        int maxiArea = 0;
    while(lt < rt){
        int hight = min(height[lt],height[rt]);
        int width = rt-lt;
        int curArea = hight*width;
        maxiArea = max(maxiArea,curArea);
        if(height[lt] < height[rt]){
            lt++;
        }
        else{
            rt--;
        }
    }
    return maxiArea;
    }
};

File: my-folder/0011-container-with-most-water/solution.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height) - 1
        maxWater = 0

        while left < right:
            width = right - left
            ht = min(height[left], height[right])
            currWater = width * ht
            maxWater = max(maxWater, currWater)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return maxWater

